### DISCUSSION TITLE
v0.7.0

### DISCUSSION THREAD

**Discussion by trollixx** [zealdocs/zeal#1541] [Announcements]
[All changes](https://github.com/zealdocs/zeal/compare/v0.6.1...v0.7.0) | [All closed issues](https://github.com/zealdocs/zeal/milestone/14?closed=1)

Finally, a new release! Only several days short of the 5 year anniversary since v0.6.1.

Thanks to everyone who was reporting issues, suggesting improvements, contributing changes, testing development builds, or just patiently waiting.

### Outline

The most significant changes are listed below.

* Web view has been migrated to Qt WebEngine (Chromium). This solves a lot of rendering issues and crashes caused by the unmaintained Qt WebKit.
* Docset pages are now served over built-in HTTP server, which addresses a number of docset specific issues.
* Emulated tabs have been replaced with real widgets, so each tab contains its own web view. Tab bar has been relocated to the top of the application window.
* Significantly improved dark mode support, including both user interface and web view (now uses native Chromium dark mode).
* Keyboard shortcuts are now more browser-like, with many new shorcuts added.
* Qt 6 support.
* CI has been migrated to GitHub Actions with Windows and Linux (AppImage) binaries provided for each build.

### Features

* add Qt 6 support ([#1420](https://github.com/zealdocs/zeal/issues/1420)) ([990be92](https://github.com/zealdocs/zeal/commit/990be92541e82e7bace2c6f31bcdd39342735279))
* **app,core,ui:** remove ads (fixes [#779](https://github.com/zealdocs/zeal/issues/779)) ([680a1ba](https://github.com/zealdocs/zeal/commit/680a1baa701581fdf5fcd92e3c2510e136666d3f))
* **app,registry:** improve index page detection, add 404 page ([c16945b](https://github.com/zealdocs/zeal/commit/c16945bc21ee94e5db55d5e77b94360770102437))
* **browser,sidebar,ui:** implement real tabs ([#1081](https://github.com/zealdocs/zeal/issues/1081)) ([0b45d6e](https://github.com/zealdocs/zeal/commit/0b45d6ebeab4943fd4ad88311e422f8190ebdf51))
* **browser,ui:** port to Qt WebEngine ([#1125](https://github.com/zealdocs/zeal/issues/1125)) ([5360da3](https://github.com/zealdocs/zeal/commit/360da3d7f34906ff41c10ac74e6b01636598903)), closes [#658](https://github.com/zealdocs/zeal/issues/658) [#776](https://github.com/zealdocs/zeal/issues/776) [#868](https://github.com/zealdocs/zeal/issues/868) [#959](https://github.com/zealdocs/zeal/issues/959) [#1119](https://github.com/zealdocs/zeal/issues/1119) [#1131](https://github.com/zealdocs/zeal/issues/1131) [#1155](https://github.com/zealdocs/zeal/issues/1155) [#1195](https://github.com/zealdocs/zeal/issues/1195)
* **browser:** add Ctrl+Enter shortcut for Highlight All option ([#1476](https://github.com/zealdocs/zeal/issues/1476)) ([80d45f7](https://github.com/zealdocs/zeal/commit/80d45f753ff3aa6125781f2bae9f8c02d9076c1b)), closes [#1467](https://github.com/zealdocs/zeal/issues/1467)
* **browser:** improve local file detection ([096dcbf](https://github.com/zealdocs/zeal/commit/096dcbf07f713e5c8ea5d59310af381727f69ba0))
* **browser:** introduce centralized web settings management ([e477c4a](https://github.com/zealdocs/zeal/commit/e477c4a0c97996915bcb6629c39ab7319e764572))
* **browser:** log blocked requests ([0c47b3b](https://github.com/zealdocs/zeal/commit/0c47b3bbb1d86fdec453d76e7067c8ac6a2bc016))
* **browser:** make zoom scaling DPI aware ([#1080](https://github.com/zealdocs/zeal/issues/1080)) ([b106938](https://github.com/zealdocs/zeal/commit/b10693888f063f9510edf01233a3acc4865a8b3a))
* **browser:** use native Chromium dark mode ([#1494](https://github.com/zealdocs/zeal/issues/1494)) ([b5ca9e7](https://github.com/zealdocs/zeal/commit/b5ca9e70dd9a8fe7b5624316cf96244802e31a4f))
* **core,ui:** add support for SOCKS5 proxies (fixes [#893](https://github.com/zealdocs/zeal/issues/893)) ([4d019ff](https://github.com/zealdocs/zeal/commit/4d019fff1f21d110beb9c3bea34b775fb9760929))
* **core:** add option to ignore SSL errors ([#1521](https://github.com/zealdocs/zeal/issues/1521)) ([9eb6169](https://github.com/zealdocs/zeal/commit/9eb6169073f0b1d653f2c2f68648a4e12ea923ef))
* **core:** enable smooth scrolling by default ([09aa69e](https://github.com/zealdocs/zeal/commit/09aa69ee6e893fa5bc7fc916e94fd19a621502aa))
* **registry,ui:** respect per docset JavaScript enablement (fixes [#999](https://github.com/zealdocs/zeal/issues/999)) ([#1003](https://github.com/zealdocs/zeal/issues/1003)) ([8bfacd7](https://github.com/zealdocs/zeal/commit/8bfacd75c43e6f95dbdb28fddbd779a2c5028ade))
* **registry:** add struct mapping for the Rust docset ([0e3e427](https://github.com/zealdocs/zeal/commit/0e3e427202639dfa1f51fdac67ae4edf5f79d79f))
* **registry:** check JS enablement in metadata (fixes [#1024](https://github.com/zealdocs/zeal/issues/1024)) ([f3bdf52](https://github.com/zealdocs/zeal/commit/f3bdf52b25ab636e1ad372e095c38a8aa220bf2f))
* **registry:** report docset version in Qt::ToolTipRole ([a8e4147](https://github.com/zealdocs/zeal/commit/a8e4147db83410702d79f204d4def69fb82d1f55)), closes [#1096](https://github.com/zealdocs/zeal/issues/1096)
* **registry:** serve docset pages over HTTP ([6dc7531](https://github.com/zealdocs/zeal/commit/6dc75311b4771277c77f185ab5b00a2aa9a6f8af)), closes [#1160](https://github.com/zealdocs/zeal/issues/1160)
* **ui:** add shortcuts for zooming webview content (fixes [#767](https://github.com/zealdocs/zeal/issues/767)) ([#1012](https://github.com/zealdocs/zeal/issues/1012)) ([241b695](https://github.com/zealdocs/zeal/commit/241b6955de905da3dd099f10b7050316ef820dbb))
* **ui:** add tree view navigation with directional keys and Alt ([03e0e1e](https://github.com/zealdocs/zeal/commit/03e0e1e05dec75ccc665f237be3bd6ae9cbca29f)), closes [#1007](https://github.com/zealdocs/zeal/issues/1007)
* **ui:** allow tree view keyboard navigation if query is empty ([6968e8e](https://github.com/zealdocs/zeal/commit/6968e8e97859feea1352ceeada06310f3c51373c))
* **ui:** implement smart selection in the search box (closes [#1033](https://github.com/zealdocs/zeal/issues/1033)) ([5ef5111](https://github.com/zealdocs/zeal/commit/5ef5111bd9577ecb6eb66000542b7e563d38d6e8))
* **ui:** load search results on selection ([3118649](https://github.com/zealdocs/zeal/commit/31186495fa55e4121ee834a63d5ffd49164ae994)), closes [#749](https://github.com/zealdocs/zeal/issues/749) [#749](https://github.com/zealdocs/zeal/issues/749)
* **ui:** provide more information for download errors ([4415920](https://github.com/zealdocs/zeal/commit/44159205090b5112941674f05d851ebc90fc6a7a)), closes [#1088](https://github.com/zealdocs/zeal/issues/1088)
* **ui:** use Fusion style on Windows 10 & 11 ([#1495](https://github.com/zealdocs/zeal/issues/1495)) ([a524735](https://github.com/zealdocs/zeal/commit/a5247351295a971252346976fbbf0524ee2754cb)), closes [#811](https://github.com/zealdocs/zeal/issues/811)

### Fixes

* **app:** fix missing app icon on Wayland ([#1130](https://github.com/zealdocs/zeal/issues/1130)) ([57908f9](https://github.com/zealdocs/zeal/commit/57908f9b45529f57f01d6f2bed76fe468eb72b27))
* **app:** properly set desktop file name ([cee4e50](https://github.com/zealdocs/zeal/commit/cee4e505e9a83e5da8a2bb1999dca51a04838201))
* **assets:** prevent duplicating application icons in GNOME dock ([#1286](https://github.com/zealdocs/zeal/issues/1286)) ([043952f](https://github.com/zealdocs/zeal/commit/043952f6fc71c874758a7552f9fd282a453e1bd0))
* **assets:** update AppStream ID to org.zealdocs.zeal ([d4d7d35](https://github.com/zealdocs/zeal/commit/d4d7d35c3e137804d583c1b33c2c0116d796fbb6))
* **browser:** always use off-the-record profile ([384d94a](https://github.com/zealdocs/zeal/commit/384d94ac664a1b83431739ad3f8083aefef826eb))
* **browser:** handle blob URL scheme for Qt older than 5.12 ([9d376c2](https://github.com/zealdocs/zeal/commit/9d376c25f7982b1680138950132411ef32ab270b)), closes [#1376](https://github.com/zealdocs/zeal/issues/1376)
* **browser:** prevent WebEngine taking away focus from searchbox ([#1322](https://github.com/zealdocs/zeal/issues/1322)) ([763edca](https://github.com/zealdocs/zeal/commit/763edca12ccd6c67e51f10891d1ced8b2510904f))
* **core,ui:** fix Qt 5.11 deprecation warnings ([#1137](https://github.com/zealdocs/zeal/issues/1137)) ([99d2cee](https://github.com/zealdocs/zeal/commit/99d2ceef5f50a5e52d76bc60f6b8327104c47719))
* **core:** bind local HTTP server to 127.0.0.1 ([753ce5b](https://github.com/zealdocs/zeal/commit/753ce5bb1f478671818287c473c4d428b1289fdb)), closes [#995](https://github.com/zealdocs/zeal/issues/995)
* **core:** create root directory before extracting files ([803db6a](https://github.com/zealdocs/zeal/commit/803db6ae34afb32bfa52ce1c0fed1df8f979ef74))
* **core:** enable support for HTTP redirects ([407e416](https://github.com/zealdocs/zeal/commit/407e4161e479d145515d99eee0e12a746aa3da5e))
* **core:** fix docset storage not created in portable build ([d34d7b5](https://github.com/zealdocs/zeal/commit/d34d7b50e034b37c9cdb138bc6474efc24645fd6))
* **core:** ignore fixed font size setting when migrating from 0.6.0 ([5362fc0](https://github.com/zealdocs/zeal/commit/5362fc0b6ac0e776b4b3921aa2a3efb72873917b)), closes [#903](https://github.com/zealdocs/zeal/issues/903) [#1054](https://github.com/zealdocs/zeal/issues/1054)
* **core:** use relative storage path in portable build by default ([b7d7e0a](https://github.com/zealdocs/zeal/commit/b7d7e0a6a9b889228218110bb2330a303b773c62))
* **registry,ui,util:** order docsets case-insensitively (fixes [#244](https://github.com/zealdocs/zeal/issues/244)) ([0a33df2](https://github.com/zealdocs/zeal/commit/0a33df25b1dc0c8be1322cb405cb620cdec932f3))
* **registry,ui:** restore docset icons on tabs ([c2acb42](https://github.com/zealdocs/zeal/commit/c2acb4241f7cdb55c41f944ad8b05b3da7b400b7)), closes [#1202](https://github.com/zealdocs/zeal/issues/1202)
* **registry:** do not truncate feed name if there is no extension ([84bf54c](https://github.com/zealdocs/zeal/commit/84bf54c1c86fb5c9841cc43ff96f97e51c2daa37)), closes [#1240](https://github.com/zealdocs/zeal/issues/1240)
* **registry:** fix regression in the index file detection ([58b9276](https://github.com/zealdocs/zeal/commit/58b92765ce8d2d0233840197e2ff777adb06b642))
* **registry:** treat docset revision as integer ([5cdad0e](https://github.com/zealdocs/zeal/commit/5cdad0ee98ad65156f1dcaca5d7dec6a89b95e99)), closes [#1532](https://github.com/zealdocs/zeal/issues/1532)
* **ui:** adjust default height for the settings dialog ([1f14668](https://github.com/zealdocs/zeal/commit/1f14668fcfc7675ceeca1f3b6f329ccedd5d6236))
* **ui:** correctly position search completer label on HiDPI display ([#1136](https://github.com/zealdocs/zeal/issues/1136)) ([7fc27d3](https://github.com/zealdocs/zeal/commit/7fc27d3d44285b3997c67415979dae05fe738abb))
* **ui:** fix compilation error with Qt 5.15 ([#1218](https://github.com/zealdocs/zeal/issues/1218)) ([064aaa0](https://github.com/zealdocs/zeal/commit/064aaa05d6a3ba4ba3cf648199a3109aba2f41fe))
* **ui:** fix crash on tab duplication ([5220b70](https://github.com/zealdocs/zeal/commit/5220b7093e0fa89ba6e164e48d0214bb1a564faa))
* **ui:** fix docset storage writable check ([de5ba7b](https://github.com/zealdocs/zeal/commit/de5ba7bd0c349c49fb5d794a8eb06faa38a8504e)), closes [#1036](https://github.com/zealdocs/zeal/issues/1036) [#1059](https://github.com/zealdocs/zeal/issues/1059)
* **ui:** fix empty page title ([31524b0](https://github.com/zealdocs/zeal/commit/31524b0cf1d7af4bc6cfa78223dcb28b450f3cba))
* **ui:** fix Qt 5.11 deprecation warning ([419ef8d](https://github.com/zealdocs/zeal/commit/419ef8de8597c90a4e79acdf1991480372f6ad8e))
* **ui:** fix RTL layout in list delegates ([#1475](https://github.com/zealdocs/zeal/issues/1475)) ([ef90f24](https://github.com/zealdocs/zeal/commit/ef90f245de7f6b7c7113581a4b587ea3ec4f948f)), closes [#1463](https://github.com/zealdocs/zeal/issues/1463)
* **ui:** improve message when storage does not exist ([#1152](https://github.com/zealdocs/zeal/issues/1152)) ([5b809d8](https://github.com/zealdocs/zeal/commit/5b809d87fe730150055787cce5cc39f7f94ec3e2))
* **ui:** make Ctrl+W the default for closing tabs ([40807fc](https://github.com/zealdocs/zeal/commit/40807fc07d72ec950c592e05e5943d253dee2cfe))
* **ui:** resolve keyboard accelerator conflict in settings (fixes [#1017](https://github.com/zealdocs/zeal/issues/1017)) ([c8f69cd](https://github.com/zealdocs/zeal/commit/c8f69cdfd0e265fff31f13cac2a1f3504c8e7a65))
* **ui:** show correct number of docsets being installed ([e977deb](https://github.com/zealdocs/zeal/commit/e977debce92d0654933f17eb153e7ea5480f45ed))
* **ui:** use cache location for temporary download files ([2f02e1f](https://github.com/zealdocs/zeal/commit/2f02e1fe2ed199e0bda4dacafd9a7db1b6e1797d)), closes [#1454](https://github.com/zealdocs/zeal/issues/1454)
* use system CA certificates for requests ([#1367](https://github.com/zealdocs/zeal/issues/1367)) ([7bd8598](https://github.com/zealdocs/zeal/commit/7bd85983872e242780c615a25c057cae1b0b483a))

<hr /><em>This discussion was created from the release <a href='https://github.com/zealdocs/zeal/releases/tag/v0.7.0'>v0.7.0</a>.</em>

**Comment by idlethread** [comment]
Thanks for the release.

Any plans to update the ubuntu PPA or flatpak images?

**Reply by trollixx** [reply]
Yes, absolutely, but will take some time as I don't use either at this point.

**Reply by mquinson** [reply]
The official Debian package got updated. I suspect that Ubuntu will catch up shortly. Probably once the Debian package migrates to the Testing part of Debian (in 2 days). https://tracker.debian.org/pkg/zeal

**Reply by mquinson** [reply]
Please don't hesitate to ping me in the future when there is a new version, so that I update the official Debian package. I normally see new versions on [my dashboard](https://qa.debian.org/developer.php?login=mquinson&comaint=yes), but I don't check it every day.

**Reply by idlethread** [reply]
Thanks for updating the deb packages. The Ubuntu development release will get it, but not the last LTS release (22.04). Either the Ubuntu PPA (https://launchpad.net/~zeal-developers/+archive/ubuntu/ppa) will need to be updated or a flatpak image will work.

/me needs to migrate to Debian unstable at some point for the development machine

**Reply by trollixx** [reply]
Thanks @mquinson!

@idlethread See #1558 for the progress on PPA. Flatpak (and Snap) might happen sooner.

**Reply by trollixx** [reply]
Flatpak has been updated thanks to @guihkx. 🎉

**Comment by oclero** [comment]
Thank you! :)
